chaco/segment_plot.py

```diff
@@ -132,6 +132,28 @@ def hittest(self, *args, **kwargs):
     def map_index(self, *args, **kwargs):
         raise NotImplementedError()
 
+    def map_screen(self, data_array):
+        """Maps an Nx2x2 array of data points into screen space and returns it
+        as an array.
+
+        Implements the AbstractPlotRenderer interface.
+        """
+        # ensure data_array is an Nx2x2 ndarray
+        data_array = np.asarray(data_array)
+        data_array = data_array.reshape(-1, 2, 2)
+
+        if len(data_array) == 0:
+            return np.empty(shape=(0, 2, 2))
+
+        x_ary, y_ary = np.transpose(data_array)
+
+        sx = self.index_mapper.map_screen(x_ary)
+        sy = self.value_mapper.map_screen(y_ary)
+        if self.orientation == "h":
+            return np.transpose(np.array((sx, sy)))
+        else:
+            return np.transpose(np.array((sy, sx)))
+
```

Comment:
Note this is effectively exactly the same as BaseXYPlot.map_screen only here we enforce Nx2x2.  Previously this worked with BaseXYPlot.map_screen as the `reshape` line wasn't there to enforce Nx2 (hence Corran's comment).  However, in the `BaseXYPlot` there was a comment explicitly saying "data array is Nx2 array" but that wasn't actually what the code assumed.

To avoid potential breaks we probably don't want to make that change to `BaseXYPlot`.
But I don't see a problem with being more specific on subclasses.  ie I think it makes sense for SegmentPlot to enforce an Nx2x2 data array? That's what it wants to be working with anyway.

`BaseXYPlot` should also be made to be explicit about the input shape.  Even if it is (..., 2) that should be clearly documented / enforced.